LabController/src/bkr/labcontroller/expire_distros.py

```diff
@@ -137,14 +147,49 @@ def main():
                            'Defaults to http://localhost:8000.')
     parser.add_option('--remove-all', default=False, action='store_true',
                       help='Remove all distros from lab controller.')
+    parser.add_option('--name', default=None,
+                      help='Remove all distros with given name. Use "%" for wildcard.')
+    parser.add_option('--family', default=None,
+                      help='Remove all distros for a given family.')
+    parser.add_option('--arch', default=None,
+                      help='Remove all distros for a given architecture. When set to "all", '
+                           'steps thru each available arch to reduce memory usage.')
     options, args = parser.parse_args()
-    try:
-        check_all_trees(options.ignore_errors,
-                        options.dry_run,
-                        options.lab_controller,
-                        options.remove_all)
-    except KeyboardInterrupt:
-        pass
+    startmsg = str("INFO: expire_distros running with --lab-controller=" + options.lab_controller)
+    for i in range(1,len(sys.argv)):
+        startmsg += ' ' + sys.argv[i]
+    print('%s' % (startmsg))
+    filter = {}
+    if options.name:
+        filter['name'] = options.name
+    if options.family:
+        filter['family'] = options.family
+
+    arch_list = []
+    if options.arch:
```

Comment:
I mean having `all` is the same as not filtering at all. We should prepare the value before and we can skip this if else completely.

Reply:
I have thought about this further and I think there is no point in having an alternative behavior between `None` arch and `all`. We are basically doing the same thing, the difference is whether we iterate over the dataset at once or not. I think we should unify this and just go per arch as doing it all at once will always be a problem on larger instances.